packages/contracts-bedrock/contracts/test/L1Block.t.sol

```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
             _l1FeeOverhead: type(uint256).max,
             _l1FeeScalar: type(uint256).max
         });
+
+        assertEq(lb.number(), type(uint64).max);
+        assertEq(lb.timestamp(), type(uint64).max);
+        assertEq(lb.basefee(), type(uint256).max);
+        assertEq(lb.hash(), keccak256(abi.encode(1)));
+        assertEq(lb.sequenceNumber(), type(uint64).max);
+        assertEq(lb.batcherHash(), bytes32(type(uint256).max));
+        assertEq(lb.l1FeeOverhead(), type(uint256).max);
+        assertEq(lb.l1FeeScalar(), type(uint256).max);
+        assertEq(lb.getL1BlockHash(type(uint64).max), keccak256(abi.encode(1)));
+    }
+
+    function test_getL1BlockHash_upperBound_l1BlockHashNumberOutOfBounds_reverts() external {
+        vm.expectRevert(L1BlockHashNumberOutOfBounds.selector);
+        lb.getL1BlockHash(2); // we set the block number to 1 in the
+        // setup so we are expecting a revert here
+    }
+
+    function test_getL1BlockHash_lowerBound_l1BlockHashNumberOutOfBounds_reverts() external {
+        depositor = lb.DEPOSITOR_ACCOUNT();
+        vm.prank(depositor);
+        lb.setL1BlockValues({
+            _number: uint64(BLOCK_HASHES_SIZE + 1),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: NON_ZERO_HASH,
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+
+        vm.expectRevert(L1BlockHashNumberOutOfBounds.selector);
+        lb.getL1BlockHash(1);
+    }
+
+    function test_getL1BlockHash_hashNotSet_l1BlockHashNumberOutOfBounds_reverts() external {
```

Comment:
This can be changed to a fuzz test where the bn passed from the fuzz test is assumed > 1. You would set the l1 block value for number 1 and asser that any other query for `bn` != 1 reverts